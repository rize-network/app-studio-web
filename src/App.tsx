import React, { Suspense } from 'react';
import { Route, Routes, useLocation } from 'react-router-dom';
import { ComponentRouter } from 'src/routes/routes';
import { View, Text } from 'app-studio';

export const App: React.FC = () => {
  const location = useLocation();
  return (
    <Suspense
      fallback={
        <View
          height="100vh"
          display="flex"
          alignItems="center"
          justifyContent="center"
        >
          <Text>Loading...</Text>
        </View>
      }
    >
      <Routes location={location}>
        {ComponentRouter.map((item, index) => (
          <Route key={index} path={item.path} element={item.element} />
        ))}
      </Routes>
    </Suspense>
  );
};

export default App;
