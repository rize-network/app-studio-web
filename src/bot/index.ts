import { Bot } from './Bot';
import { DocuCode } from './DocuCode';

async function main() {
  const projectName = 'MDX Documentation';
  const descriptionPath = 'docs/description.md';

  // Get componentName from command line arguments
  const componentName = process.argv[2]; // Assuming it's the first argument passed
  if (!componentName) {
    console.error('Please provide a component name as an argument.');
    process.exit(1);
  }

  console.log(`Generating documentation for component: ${componentName}`);

  const basePath = 'src/components';
  const propsPath = `src/data/props/${componentName}.props.json`;
  const componentFolder = `${basePath}/${componentName}`;
  const componentPath = `${basePath}/${componentName}/${componentName}`;

  const assistantGPT = new Bot(componentName);

  // Implementation for Documentation of the code
  console.log('Initializing documentation process...');
  const docuCode = new DocuCode(componentName);
  const assistantDocumentation = await assistantGPT.init(
    projectName,
    [],
    descriptionPath,
    'docu'
  );

  await docuCode.processDirectory(componentFolder, assistantDocumentation.id);

  // Generating props file for code
  console.log('Generating props file...');
  const { fileIds } = await assistantGPT.addFiles(componentPath);

  const assistantCreation = await assistantGPT.init(
    projectName,
    fileIds,
    descriptionPath,
    'props'
  );

  await assistantGPT.response(assistantCreation.id, propsPath);

  await assistantGPT.MarkdownGeneration(
    componentFolder,
    componentName,
    propsPath
  );

  console.log('Documentation generation completed.');
}

main().catch((error) => {
  console.error('Error during the execution of main:', error);
});
