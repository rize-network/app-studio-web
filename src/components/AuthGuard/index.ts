export { AuthGuard, withAuthGuard, RouteAuthGuard, AuthStatus } from './AuthGuard';
