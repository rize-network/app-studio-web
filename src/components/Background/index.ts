export { Background } from './Background';
export type {
  BackgroundProps,
  AuroraBackgroundProps,
  MeteorsProps,
  WallProps,
  ParticlesProps,
  GridProps,
  RipplesProps,
  BackgroundImageProps,
  BackgroundVideoProps,
  BackgroundGradientProps,
  BackgroundStyles,
} from './Background/Background.props';
export type {
  BackgroundEffect,
  AuroraDirection,
  MeteorTiming,
  BorderAnimationStyles,
  StrokeAnimationStyles,
  BackgroundContextType,
  BackgroundSize,
  BackgroundPosition,
  BackgroundRepeat,
  BlendMode,
} from './Background/Background.type';
