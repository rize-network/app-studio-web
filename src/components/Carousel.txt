'use client';
import React, { useEffect, useRef, useState, createContext } from 'react';
import { Horizontal, Vertical } from './Layout';
import { IconButtons } from './Button/examples';
import { Button } from './Button/Button';
import { RightArrowSvg } from 'src/components/Svg/index';

// import { IconLeft, IconRight } from '@app-studio/web/icons';

interface CarouselProps {
  items: JSX.Element[];
  initialScroll?: number;
}

export const CarouselContext = createContext({
  onCardClose: (index: number) => {},
  currentIndex: 0,
});

export const Carousel = ({ items, initialScroll = 0 }: CarouselProps) => {
  const carouselRef = useRef<HTMLDivElement>(null);
  const [canScrollLeft, setCanScrollLeft] = useState(false);
  const [canScrollRight, setCanScrollRight] = useState(true);

  useEffect(() => {
    if (carouselRef.current) {
      carouselRef.current.scrollLeft = initialScroll;
      checkScrollability();
    }
  }, [initialScroll]);

  const checkScrollability = () => {
    if (carouselRef.current) {
      const { scrollLeft, scrollWidth, clientWidth } = carouselRef.current;
      setCanScrollLeft(scrollLeft > 0);
      setCanScrollRight(scrollLeft < scrollWidth - clientWidth);
    }
  };

  const scrollLeft = () =>
    carouselRef.current?.scrollBy({ left: -300, behavior: 'smooth' });
  const scrollRight = () =>
    carouselRef.current?.scrollBy({ left: 300, behavior: 'smooth' });

  return (
    <CarouselContext.Provider
      value={{ onCardClose: () => {}, currentIndex: 0 }}
    >
      <Horizontal
        gap={2}
        alignItems="center"
        width="100%"
        paddingVertical={20}
        position="relative"
      >
        <Button
          onClick={scrollLeft}
          icon={<RightArrowSvg />}
          disabled={!canScrollLeft}
          shape="rounded"
        />
        <Horizontal
          ref={carouselRef}
          onScroll={checkScrollability}
          gap={4}
          overflowX="auto"
          paddingX={10}
        >
          {items.map((item, index) => (
            <Vertical key={index} minWidth="300px">
              {item}
            </Vertical>
          ))}
        </Horizontal>
        <Button
          onClick={scrollRight}
          icon={<RightArrowSvg />}
          disabled={!canScrollRight}
          shape="rounded"
        />
      </Horizontal>
    </CarouselContext.Provider>
  );
};
