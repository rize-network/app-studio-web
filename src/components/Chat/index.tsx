/**
 * App Studio Components for AI Applications
 *
 * This module exports a collection of components specifically designed
 * for building AI-powered applications.
 */

// Core components
export * from './ChatInterface/ChatInterface';
export * from './MessageBubble/MessageBubble';
export * from './PromptInput/PromptInput';
export * from './ModelSelector/ModelSelector';
export * from './AIResponseDisplay/AIResponseDisplay';
export * from './LoadingIndicator/LoadingIndicator';

// Supporting components
export * from './ToolSelector/ToolSelector';
export * from './SettingsPanel/SettingsPanel';
export * from './FileUploader/FileUploader';
export * from './CodeBlock/CodeBlock';
