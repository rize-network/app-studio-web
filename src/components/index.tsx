export * from './Button/Button';
export * from './Form/Checkbox/Checkbox';
export * from './Form/CountryPicker/CountryPicker';
export * from './Form/DatePicker/DatePicker';
export * from './Form/Password/Password';
export * from './Form/Select/Select';
export * from './Form/Switch/Switch';
export * from './Form/Switch/Switch';
export * from './Form/TextArea/TextArea';
export * from './Form/TextField/TextField';
export * from './Layout/Center/Center';
export * from './Formik';
export * from './Layout/Horizontal/Horizontal';
export * from './Layout/Vertical/Vertical';
export * from './Layout/View/View';
export * from './Link/Link';
export * from './Loader/Loader';
export * from './Modal/Modal';
export * from './Text/Text';
export * from './File/File';
export * from './Tabs/Tabs';
export * from './Alert/Alert';
export * from './AspectRatio/AspectRatio';
export * from './Avatar/Avatar';
export * from './Badge/Badge';
export * from './Toggle/Toggle';
export * from './ToggleGroup/ToggleGroup';
export * from './Message/Message';
