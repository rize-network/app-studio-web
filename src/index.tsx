import React from 'react';
import ReactDOM from 'react-dom/client';
import AppProvider from 'src/providers';
import { App } from './App';
import { ComboBoxProvider } from './components/Form/ComboBox/ComboBox/ComboBox.provider';

const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement
);
root.render(
  <React.StrictMode>
    <AppProvider>
      <ComboBoxProvider>
        <App />
      </ComboBoxProvider>
    </AppProvider>
  </React.StrictMode>
);
