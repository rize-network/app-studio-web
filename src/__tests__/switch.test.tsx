import React from 'react';
import renderer from 'react-test-renderer';
import { Switch } from 'src/components';

import { cleanup, render, screen } from '@testing-library/react';

afterEach(() => {
  cleanup();
});

// test('renders Switch component', () => {
//   render(<Switch name="active" role="textbox" />);
//   const SwitchElement = screen.getByRole('textbox');
//   expect(SwitchElement).toBeInTheDocument();
// });

// test('renders disabled Switch', () => {
//   render(<Switch name="active" isDisabled role="textbox" />);
//   const inputElement = screen.getByRole('textbox');
//   expect(inputElement).toBeDisabled();
// });

test('Switch to match snapshot', () => {
  const tree = renderer
    .create(
      <Switch
        name="name"
        size="xl"
        views={{
          slider: {
            backgroundColor: 'red',
          },
          circle: {
            backgroundColor: 'blue',
          },
        }}
        shadow={{ boxShadow: 'rgb(249, 115, 22) 0px 4px 14px 0px' }}
        isReadOnly
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
