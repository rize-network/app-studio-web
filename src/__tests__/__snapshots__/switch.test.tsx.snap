// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch to match snapshot 1`] = `
<label
  className="sc-bdfCDU krrJWc"
  fontSize="sm"
  fontStyle="normal"
  fontWeight={400}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  textDecoration="none"
  width="100%"
>
  <input
    checked={false}
    className="sc-bdfCDU krrJWc"
    disabled={false}
    height={0}
    name="name"
    onChange={[Function]}
    onClick={[Function]}
    opacity={0}
    readOnly={true}
    type="checkbox"
    width={0}
  />
  <div
    alignItems="center"
    backgroundColor="red"
    borderRadius={24}
    boxShadow="rgb(249, 115, 22) 0px 4px 14px 0px"
    className="sc-bdfCDU fxLFlD"
    cursor="pointer"
    display="flex"
    filter="brightness(1)"
    height={32}
    justifyContent="flex-start"
    onClick={[Function]}
    paddingHorizontal={8}
    paddingVertical={0}
    transition="justify-content 0.3s cubic-bezier(0.4, 0, 0.2, 1)"
    width={60}
  >
    <div
      backgroundColor="blue"
      borderRadius="50%"
      className="sc-bdfCDU KifEJ"
      height={18}
      onClick={[Function]}
      width={18}
    />
  </div>
</label>
`;
